app/src/main/java/com/gholem/moneylab/features/planning/viewmodel/PlanningViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.gholem.moneylab.features.planning.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.gholem.moneylab.arch.nav.NavigationLiveData
+import com.gholem.moneylab.domain.model.TransactionModel
+import com.gholem.moneylab.features.add.domain.InsertTransactionsModelUseCase
+import com.gholem.moneylab.features.planning.domain.FetchTransactionModelUseCase
+import com.gholem.moneylab.features.planning.navigation.PlanningNavigationEvent
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import timber.log.Timber.i
+import javax.inject.Inject
+
+@HiltViewModel
+class PlanningViewModel @Inject constructor(
+    private val insertTransactionsModelUseCase: InsertTransactionsModelUseCase,
+    private val fetchTransactionModelUseCase: FetchTransactionModelUseCase
+) : ViewModel() {
+
+    private val _actions = Channel<Action>(Channel.BUFFERED)
+    val actions = _actions.receiveAsFlow()
+
+    val navigation: NavigationLiveData<PlanningNavigationEvent> =
+        NavigationLiveData()
+
+    fun fetchTransactions() = viewModelScope.launch {
+        val transactions = fetchTransactionModelUseCase.run(Unit)
+        Action.ShowTransactions(transactions).send()
+    }
+
+    fun navigateToCategoryBottomSheet() = viewModelScope.launch {
+        navigation.emit(PlanningNavigationEvent.ToChartScreen)
+    }
+
+    fun saveNewTransactionFromPerson(positon: Int) = viewModelScope.launch {
+        val transactions = fetchTransactionModelUseCase.run(Unit)
+        insertTransactionsModelUseCase.run(listOf(
```

Comment:
Also as I said on the call - make new UseCase to save single object instead of using UseCase which saves a list of objects